core/src/executor/evaluate/error.rs

```diff
@@ -61,3 +64,60 @@ pub enum EvaluateError {
     #[error("format function does not support following data_type: {0}")]
     UnsupportedExprForFormatFunction(String),
 }
+
+#[derive(Error, Serialize, Debug, PartialEq)]
+pub enum ChronoFormatError {
+    /// Given field is out of permitted range.
+    #[error("given field is out of permitted range")]
+    OutOfRange,
+
+    /// There is no possible date and time value with given set of fields.
+    ///
+    /// This does not include the out-of-range conditions, which are trivially invalid.
+    /// It includes the case that there are one or more fields that are inconsistent to each other.
+    #[error("the given date and time value is impossible to be formmated")]
+    Impossible,
+
+    /// Given set of fields is not enough to make a requested date and time value.
+    ///
+    /// Note that there *may* be a case that given fields constrain the possible values so much
+    /// that there is a unique possible value. Chrono only tries to be correct for
+    /// most useful sets of fields however, as such constraint solving can be expensive.
+    #[error("given set of field is not enough to be formatted")]
+    NotEnough,
+
+    /// The input string has some invalid character sequence for given formatting items.
+    #[error("given format string has invalid specifier")]
+    Invalid,
+
+    /// The input string has been prematurely ended.
+    #[error("input string has been permaturely ended")]
+    TooShort,
+
+    /// All formatting items have been read but there is a remaining input.
+    #[error("given format string is missing some specifier")]
+    TooLong,
+
+    /// There was an error on the formatting string, or there were non-supported formating items.
+    #[error("given format string includes non-supported formmating item")]
+    BadFormat,
+
+    // TODO: Change this- to `#[non_exhaustive]` (on the enum) when MSRV is increased
+    #[error("unreachable chrono format error")]
+    Unreachable,
+}
+
+impl From<chrono::format::ParseError> for ChronoFormatError {
+    fn from(error: chrono::format::ParseError) -> ChronoFormatError {
+        match error.kind() {
+            chrono::format::ParseErrorKind::OutOfRange => ChronoFormatError::OutOfRange,
+            chrono::format::ParseErrorKind::Impossible => ChronoFormatError::Impossible,
+            chrono::format::ParseErrorKind::NotEnough => ChronoFormatError::NotEnough,
+            chrono::format::ParseErrorKind::Invalid => ChronoFormatError::Invalid,
+            chrono::format::ParseErrorKind::TooShort => ChronoFormatError::TooShort,
+            chrono::format::ParseErrorKind::TooLong => ChronoFormatError::TooLong,
+            chrono::format::ParseErrorKind::BadFormat => ChronoFormatError::BadFormat,
+            chrono::format::ParseErrorKind::__Nonexhaustive => ChronoFormatError::Unreachable,
```

Comment:
Repeated `chrono::format::ParseErrorKind` may reduce readability.

```Suggestion
           Use chrono::format::ParseErrorKind::*;

           error.kind() match {
               OutOfRange => ChronoFormatError::OutOfRange,
               Impossible => ChronoFormatError:: Impossible,
               NotEnough => ChronoFormatError::NotEnough,
               invalid => ChronoFormatError::invalid,
               TooShort => ChronoFormatError::TooShort,
               TooLong => ChronoFormatError::TooLong,
               BadFormat => ChronoFormatError::BadFormat,
               __Nonexhaustive => ChronoFormatError::unreachable,
           }
```
How about this?

Reply:
I totally agree with this